core/deployment/src/main/java/io/quarkus/deployment/ExtensionLoader.java

```diff
@@ -733,8 +744,14 @@ public boolean canHandleObject(final Object obj, final boolean staticInit) {
                             if (isRecorder && phase == ConfigPhase.BUILD_AND_RUN_TIME_FIXED) {
                                 runTimeProxies.computeIfAbsent(parameterClass, readResult::requireRootObjectForClass);
                             }
-                        } else if (phase == ConfigPhase.RUN_TIME) {
+                        } else if (phase == ConfigPhase.BOOTSTRAP || phase == ConfigPhase.RUN_TIME) {
                             if (isRecorder) {
+                                if ((phase == ConfigPhase.BOOTSTRAP)
+                                        && !method.getReturnType().equals(RunTimeConfigurationSourceValueBuildItem.class)) {
```

Comment:
I agree completely. The whole reason I did it this way is because I wanted to be able to distringuish the bootstrap recorders for the regular ones and therefor generate bytecode that interacts with Config properly.
If I can change the PR to leverage the Build Steps that you mention in order comments, than this and other comments on the PR will likely go away.

Reply:
This arbitrary restriction has now been lifted.